hypershift-operator/controllers/platform/aws/controller.go

```diff
@@ -42,9 +47,38 @@ type AWSEndpointServiceReconciler struct {
 	elbv2Client elbv2iface.ELBV2API
 }
 
+func mapNodePoolToAWSEndpointServicesFunc(c client.Client) func(obj client.Object) []reconcile.Request {
+	return func(obj client.Object) []reconcile.Request {
+		nodePool, ok := obj.(*hyperv1.NodePool)
+		if !ok {
+			return []reconcile.Request{}
+		}
+
+		// This is a pretty fragile but without a client or context with which to list the
```

Comment:
An alternate would be to use a client and list the endpointservices here, the main drawback is that we can't handle errors

Reply:
Yeah, I thought of this and you still don't avoid a fragile mapping in that the HCP namespace has to be assumed from the nodepool Namespace and the node pool clusterName.  All in all, it added complexity and was still fragile.

Reply:
>I thought of this but we still don't avoid a fragile mapping in that the HCP namespace has to be assumed from the nodepool Namespace and the node pool clusterName

Could fetch HC from nodePool.spec.clusterName.
Then `HCP_namespace(HC)` is a stronger contract.
Then list endpoints.
?

That said I think I lean towards the simplicity of current approach.
It'd be good though to document this 2 dependents aws endpoints controllers at some point may be here https://hypershift-docs.netlify.app/reference/controller-architecture/

Reply:
Sounds like this is acceptable for now